spec/ui/command_line_spec.moon

```diff
@@ -0,0 +1,276 @@
+-- Copyright 2012-2015 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+import app, dispatch, interact from howl
+import Window from howl.ui
+
+describe 'commandline', ->
+  local command_line, within_activity
+
+  run_in_coroutine = (f) ->
+    wrapped = coroutine.wrap -> f!
+    return wrapped!
+
+  before_each ->
+    app.window = Window!
+    command_line = app.window.command_line
+    within_activity = (f) ->
+      command_line\run
+        name: 'within-activity'
+        handler: -> f!
+
+  after_each ->
+    ok, result = pcall -> app.window.command_line\abort_all!
+    if not ok
+      print result
+    within_activity = nil
+    command_line = nil
+    app.window = nil
+
+  describe 'command_line', ->
+    describe '\\run(activity_spec)', ->
+      it 'errors if handler or factory field not in spec', ->
+        f = -> command_line\run {}
+        assert.has_error f, 'activity_spec requires "name" and one of "handler" or "factory" fields'
```

Comment:
I see that the standard error assert takes two arguments now, but I think that's in upcoming 2 release of busted?